pkg/controller/client.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"google.golang.org/grpc"
+
+	"kmesh.net/kmesh/pkg/controller/config"
+	"kmesh.net/kmesh/pkg/controller/envoy"
+	"kmesh.net/kmesh/pkg/controller/workload"
+	"kmesh.net/kmesh/pkg/nets"
+)
+
+const (
+	RandTimeSed = 1000
+)
+
+type XdsClient struct {
+	ctx            context.Context
+	cancel         context.CancelFunc
+	grpcConn       *grpc.ClientConn
+	client         service_discovery_v3.AggregatedDiscoveryServiceClient
```

Comment:
This field has been encapsulated into `adsStream` and `workloadStream`, maybe we no longer need to keep it

Reply:
When reconnection occurs, it is necessary to create a new client.